pkg/tarmak/stack/rules.go

```diff
@@ -0,0 +1,362 @@
+// Copyright Jetstack Ltd. See LICENSE for details.
+package stack
+
+import (
+	"fmt"
+)
+
+type Rules struct{}
+
+type FirewallRule struct {
+	Name        string
+	Type        string
+	FromPort    int
+	ToPort      int
+	Protocol    string
+	CIDIRBlocks []string
+}
+
+func FirewallRules() *Rules {
+	return &Rules{}
+}
+
+func (r *Rules) VaultRules() []*FirewallRule {
+	return []*FirewallRule{
+		r.Rule_vault_out_allow_all(),
+		r.Rule_vault_in_allow_ssh_bastion(),
+		r.Rule_vault_in_allow_vault_bastion(),
+		r.Rule_vault_in_allow_everything_inner_cluster(),
+	}
+}
+
+func (r *Rules) BationRules(adminCIRD string) []*FirewallRule {
+	return []*FirewallRule{
+		r.Rule_egress_allow_all(),
+		r.Rule_ingress_allow_ssh_all(adminCIRD),
+	}
+}
+
+func (r *Rules) RemoteAdminRules() []*FirewallRule {
+	return []*FirewallRule{
+		r.Rule_remote_admin_egress_allow_all(),
+		r.Rule_remote_admin_ingress_allow_ssh_all(),
+	}
+}
+
+func (r *Rules) PoolRules(poolName string) []*FirewallRule {
+	return []*FirewallRule{
+		r.Rule_vault_allow_vault_from_pool(poolName),
+		r.Rule_bastion_allow_wing_from_pool(poolName),
+		r.Rule_pool_allow_ssh_from_bastion(poolName),
+		r.Rule_pool_allow_all_from_pool(poolName),
+		r.Rule_pool_allow_egress(poolName),
+	}
+}
+
+func (r *Rules) ELBRules(poolName string) []*FirewallRule {
+	return []*FirewallRule{
+		r.Rule_pool_allow_api_from_elb(poolName),
+		r.Rule_pool_elb_allow_api_from_kube_worker(poolName),
+		r.Rule_pool_elb_allow_api_from_bastion(poolName),
+		r.Rule_pool_allow_api_from_pool(poolName),
+		r.Rule_pool_elb_allow_api_to_pool(poolName),
+	}
+}
+
+func (r *Rules) EtcdRules(poolName string) []*FirewallRule {
+	return []*FirewallRule{
+		r.Rule_pool_allow_etcd_overlay_from_kubernetes_worker(poolName),
+		r.Rule_pool_allow_etcd_overlay_from_kubernetes_master(poolName),
+		r.Rule_pool_allow_etcd_k8sevents_from_kubernetes_master(poolName),
+	}
+}
+
+func (r *Rules) WorkerRules(poolName string) []*FirewallRule {
+	return []*FirewallRule{
+		r.Rule_pool_allow_all_from_kubernetes_master(poolName),
+	}
+}
+
+func (r *Rules) MasterRules(poolName string) []*FirewallRule {
+	return []*FirewallRule{
+		r.Rule_pool_allow_bgp_from_kube_worker(poolName),
+		r.Rule_pool_allow_ipip_from_kuber_worker(poolName),
+		r.Rule_pool_allow_api_from_kube_woker(poolName),
+	}
+}
```

Comment:
You should be able to replace the bunch of these methods  by some lookup method, that finds rules which have `Role == 'bastion'` in `Sources` instead of `BationRules`

Services for all can be defined as global var